### ISSUE TITLE
Feature Request: Pass the clicked element and the scroll-to target element in `before()` and `after()` callbacks

### ISSUE THREAD

**Issue by chandlervdw** [cferdinandi/smooth-scroll#395]
DOM elements would be rad
```
const scroll = new SmoothScroll('a[data-scroll]', {
  before: (clickTarget, scrollTarget) => { },
  after: (clickTarget, scrollTarget) => { }
})
```

**Comment by chandlervdw** [comment]
Maybe I missed the documentation but on a click event, the `scrollTarget` is the first arg and the `clickTarget` is the second arg. In an `animateScroll` event, the `scrollTarget` if the first arg and the options are the second arg. Correct me if I'm wrong!

**Comment by cferdinandi** [comment]
It does, but I apparently dropped it out of the docs by accident. I'll get that fixed!

**Comment by zipper** [comment]
It would also be useful to know the clicked element in `offset` function. I need to change the offset dynamically based on clicked element and don't want to create new instance of smooth scroll because of one case.

**Comment by cferdinandi** [comment]
Docs updated to reflect the `before` and `after` callback arguments.

@zipper Good idea! Please open a new issue with this feature request. Thanks!
